src/test/java/io/cdap/plugin/ariba/source/AribaBatchSourceTest.java

```diff
@@ -189,7 +192,7 @@ public void testValidateAdvancedParametersError() {
       List<ValidationFailure> failures = s.getFailures();
       Assert.assertEquals("Failures size does not match, " +
                             "'CDF_ARIBA_01501 - Failed to call given Ariba service.'",
-                          1, failures.size());
+                          3, failures.size());
```

Comment:
why asserting size to `3` and not `1`?

Reply:
Acc to the previous code if credentials are incorrect it was throwing single error for all the incorrect credentials fields. But now the failures are being segregated, so it is throwing 3 failures for clientId, clientSecret and apiKey.